pilot/pkg/features/pilot.go

```diff
@@ -552,6 +552,16 @@ var (
 			"Regardless of this setting, the configuration can be overridden with the Sidecar.Ingress.DefaultEndpoint configuration.",
 	).Get()
 
+	// EnableHBONE provides a global Pilot flag for enabling HBONE.
+	// Generally, this could be a per-proxy setting (and is, via ENABLE_HBONE node metadata).
+	// However, there are some code paths that impact all clients, hence the global flag.
+	// Warning: do not enable by default until endpoint_builder.go caching is fixed (and possibly other locations).
+	EnableHBONE = env.Register(
```

Comment:
Not sure I follow - why do we need 2 settings to enable ? One injection setting ( using META_..) or a label should be enough. 

I think it would be ideal to enable this from the workload side, not centrally.

Reply:
That was my original plan but I added this because the endpoint_builder part requires disabling a critical cache in order to support HBONE; this cache is not workload-bound.

I would hope we find a way around that issue and remove this setting shortly after